Team B, prototype pull request

skim451 opened the pull request:
박예준, 이신은, 김세환 조

- 배민찬 MainPage UI를 만들었습니다. 
- 자동완성 기능을 추가하였습니다.

kkiluk committed:
파일 생성

html, css, json 파일 생성했습니다.

kkiluk committed:
appbar.html 구조 만들다.

appbar.html 구조와 css 기본을 구현했습니다.
이미지 리소스 등록했습니다.

skim451 committed:
switch coder

smartyjpark committed:
Modify StyleSheet

Modify style.css and appbar.html

skim451 committed:
Added header-wrapper and mouse hovering

added header-wrapper, so the contents stay at the center 980 px.
added mouse hovering properties for a tag elements: color -> mint,
cursor -> pointer.

kkiluk committed:
js 클래스 구조를 생성하고 자동완성 로직을 작성하다.

js 모든 클래스 구조를 생성했습니다.
자동완성 로직을 작성했습니다.

smartyjpark committed:
Create autocomplete list window and Function

검색어에 따른 자동완성 리스트 DOM 및 기능 구현
키워드 하이라이팅 기능 구현
Rendering / Hovering 기능 추가

skim451 committed:
Added keyboardevent and mouse click event for autocomplete list

search-field now supports arrow up/down and enter events.
Auto-complete list also has a mouse click event, which adds
selected item to the search field.

skim451 committed:
Completed auto complete functions

Added mouse hovering for auto complete list.

skim451 committed:
Fixed a minor bug

Changed keyboard event listener from 'keypress' to 'keydown'.
Added pointer type cursor to auto complete list mouse hovering.

kkiluk committed:
Share issue

html inside html

kkiluk committed:
side-dish의 정렬 문제를 공유하다.

side-dish-title과 thumb-imgs를 세로로 정렬하고 싶습니다.

kkiluk committed:
Fixed 스타일 and Added focusout 이벤트

top-navigation의 글씨색을 수정했습니다.
search-bar의 css를 수정했습니다.
menu-bar의 글씨 크기와 볼트 처리를 적용했습니다.
브랜드관의 배경색을 변경했습니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+	}
+}
+
+class MainpageRenderer {
```
?

skim451 replied:
원래 계획은 js에서 동적으로 렌더링 해주는 것이었는데... (슬라이딩 UI까지 고려하여...) 
당장은 html에서 다 해주기 때문에 버렸습니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+	}
+}
+
+class AjaxRequest {
```
이녀석은 class를 가질만한 크기는 아니니까, 다른 클래스내 함수로 만들어도 될 거 같네요.

skim451 replied:
Utility Class를 만들어 넣었습니다.
$() 같은 전역함수를 같이 묶었습니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+	return document.querySelector(ele);
+}
+
+class DomContainer {
```
DomContainer를 만들어서 어떤 이득이 있는거 같아요?

skim451 replied:
querySelector를 매번 부를 필요가 없습니다. 
dom에 대한 접근을 DomContainer를 통해 하여 Dom에 대한 접근을 관리하고자 하였습니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+			const value = returnData[1];
+
+			this.memo[key] = value;
+			if (!this.memoLog.includes(key)) {
```
유용한 메서드들이 가끔 브라우저 지원범위가 좀 안좋아요.
includes 의 브라우저 지원범위를 한번 살펴보세요.
그리고 polyfill코드를 찾아보고, 서비스에 polyfill코드를 추가해보죠.

skim451 replied:
include polyfill 추가했습니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+		searchField.addEventListener('keydown', function(e) {
+			let currHoveredItem = this.domContainer.getHoveredItem(); 
+
+			switch(e.keyCode){
```
switch 문은 없애서 코딩한다면 어떻게 해야할까요?

skim451 replied:
객체 리터럴로 바꾸는 방법을 숙지하였으나, 
현재는 스위치를 쓰는게 더 가시성이 좋다고 생각됩니다.

crongro commented on app.js:
```diff
@@ -0,0 +1,245 @@
+function $(ele) {
```
전역공간에 덩그러니 있는 이녀석을 어떻게 하는게 좋을지 생각해보세요.

skim451 replied:
Utility Class를 만들어 넣었습니다.

crongro reviewed (approved):
리뷰 내용을 고민하고, 일부분 적용하신후에 다음단계로 가보죠.

skim451 committed:
Applied feedbacks given by crong

1. Added polyfill for include()
2. Removed AppBarRenderer and MainpageRenderer
3. Merged $() and AjaxRequest into Util Class

crongro commented on app.js:
```diff
@@ -1,63 +1,74 @@
-        	listDom.innerHTML = ""
-        	return false
+//ployfill
+if (!Array.prototype.includes) {
```
polyfill은 별도 파일로 묶어서 따로 관리해도 좋을 거에요.
